src/runtime/recipe/recipe-resolver.ts

```diff
@@ -89,19 +104,25 @@ export class ResolveWalker extends RecipeWalker {
       return true;
     });
 
-    if (mappable.length === 1) {
-      return (recipe, handle) => {
-        handle.mapToStorage(mappable[0]);
-        return 0;
-      };
+    if (mappable.length === 0) {
+      return error('No mappable connections (due to filtering)');
     }
-    return undefined;
+    return mappable.map(store => ((recipe, updateHandle) => {
+      updateHandle.mapToStorage(store);
+      return 0;
+    }));
   }
 
-  onSlotConnection(_recipe: Recipe, slotConnection: SlotConnection) {
+  onSlotConnection(_recipe: Recipe, slotConnection: SlotConnection): Continuation<Recipe, SlotConnection[]> {
+    const error = (label: string) => {
+      if (this.options && this.options.errors) {
+        this.options.errors.set(slotConnection, label);
+      }
+      return [];
+    };
     const arc = this.arc;
     if (slotConnection.isConnected()) {
-      return undefined;
+      return error('Slot connection is already connected');
```

Comment:
Why is it an error?

Reply:
I think it is because there is no way to change the current recipe to connect the slot (as it is already connected).
This is not a new error, just a new message for the error.